nodeup/pkg/model/kubeapiserver.go

```diff
@@ -127,22 +141,30 @@ func (b *KubeAPIServerBuilder) writeAuthenticationConfig(c *fi.ModelBuilderConte
 			Type:     nodetasks.FileType_File,
 		}
 		c.AddTask(t)
+
 		return nil
-	} else {
-		return fmt.Errorf("Unrecognized authentication config %v", b.Cluster.Spec.Authentication)
 	}
+
+	return fmt.Errorf("Unrecognized authentication config %v", b.Cluster.Spec.Authentication)
```

Comment:
i guess the a idiom would to be return early ... i.e. b.Cluster.Spec.Authentication.Kopeio == nil then return the error .. else everything continues